source/LocationFilter.cpp

```diff
@@ -215,6 +212,31 @@ bool LocationFilter::Matches(const Ship &ship) const
 
 
 
+// Convert a distance filter into a "near" filter.
+LocationFilter LocationFilter::SetOrigin(const System *origin) const
+{
+	if(IsEmpty() || originMaxDistance < 0)
+		return *this;
+	
+	// Copy all parts of this instantiated filter into the result.
+	LocationFilter result = *this;
+	// If this LocationFilter has already defined a "near <system>" filter, then
+	// do not convert a "distance" filter into a "near" filter.
+	if(!center && originMaxDistance > -1)
```

Comment:
Since `originMaxDistance` is an integer, `originMaxDistance > -1` is equivalent to `!(originMaxDistance < 0)`, so it will always be true if you get to this point in the function.